src/sidebar/util/time.js

```diff
@@ -224,23 +217,46 @@ export function decayingInterval(date, callback) {
 }
 
 /**
- * This callback is a param for the `decayingInterval` function.
- * @callback UpdateCallback
- * @param {string} date - The date associated with the current interval/timeout
- */
-
-/**
- * Formats a date as a string relative to the current date.
+ * Formats a date as a short approximate string relative to the current date.
+ *
+ * The level of precision is proportional to how recent the date is.
+ *
+ * For example:
+ *
+ *  - "Just now"
+ *  - "5 minutes ago"
+ *  - "25 Oct 2018"
  *
  * @param {Date|null} date - The date to consider as the timestamp to format.
  * @param {Date} now - The date to consider as the current time.
- * @param {Object} [Intl] - JS internationalization API implementation; this
- *                      param is present for dependency injection during test.
+ * @param {Intl} [Intl] - Test seam. JS `Intl` API implementation.
  * @return {string} A 'fuzzy' string describing the relative age of the date.
  */
-export function toFuzzyString(date, now, Intl) {
+export function formatRelativeDate(date, now, Intl) {
   if (!date) {
     return '';
   }
-  return getBreakpoint(date, now).formatFn(date, now, Intl);
+  return getBreakpoint(date, now).formatter(date, now, Intl);
+}
+
+/**
+ * Formats a date as an absolute string in a human readable format.
+ *
+ * The exact format will vary depending on the locale, but the verbosity will
+ * be consistent across locales. In en-US for example this will look like:
+ *
+ *  "Sunday, Dec 17, 2017, 10:00 AM"
+ *
+ * @param {Date} date
+ * @return {string}
+ */
+export function formatDate(date) {
+  return format(date, {
+    year: 'numeric',
+    month: 'short',
+    day: '2-digit',
+    weekday: 'long',
+    hour: '2-digit',
+    minute: '2-digit',
+  });
 }
```

Comment:
I suggest to add a `Int` test seam here as well.

```suggestion
 * @param {Intl} [Intl] - Test seam. JS `Intl` API implementation.
 * @return {string}
 */
export function formatDate(date, Intl) {
  return format(
    date,
    {
      year: 'numeric',
      month: 'short',
      day: '2-digit',
      weekday: 'long',
      hour: '2-digit',
      minute: '2-digit',
    },
    Intl
  );
}
```